DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/DataSourceProvider.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Microsoft.EntityFrameworkCore;
+
+namespace DfE.FindInformationAcademiesTrusts.Data.AcademiesDb;
+
+public class DataSourceProvider : IDataSourceProvider
+{
+    private readonly IAcademiesDbContext _academiesDbContext;
+
+    [ExcludeFromCodeCoverage]
+    public DataSourceProvider(AcademiesDbContext academiesDbContext) : this(
+        (IAcademiesDbContext)academiesDbContext)
+    {
+    }
+
+    public DataSourceProvider(IAcademiesDbContext academiesDbContext)
+    {
+        _academiesDbContext = academiesDbContext;
+    }
+
+    public async Task<DataSource?> GetGiasUpdated()
+    {
+        var lastEntry = await _academiesDbContext.ApplicationEvents
+            .Where(e => e.Source != null
+                        && e.Source.Contains("adf-t1") && e.Source.Contains("-sips-dataflow")
+                        && e.Message == "Finished"
+                        && e.EventType != 'E'
+                        && e.Description == "GIAS_Daily").MaxAsync(e => e.DateTime);
+        if (lastEntry is null) return null;
+        return new DataSource("Get information about schools", lastEntry.Value, "Daily");
```

Comment:
`"Get information about schools"` and `"Daily"` are UI display names. Here in the backend it would be better to use enums as it's easier to process, less likely to fall victim to typos and far quicker to change if we decide one day to change the UI code to say something like "every day"